extensions/interactions/LogicProof/LogicProof.js

```diff
@@ -109,33 +130,47 @@ oppia.directive('oppiaInteractiveLogicProof', [
             // We update the message only if the user has added or removed a
             // line break, so that it remains while they work on a single line.
             if (change.text.length > 1 || change.removed.length > 1) {
-              $scope.refreshMessages(editor);
+              $scope.checkForBasicErrors();
             }
           });
+
+          $scope.editor = editor;
         };
 
-        $scope.refreshMessages = function(editor) {
-          if($scope.mistakeMark) {
-            $scope.mistakeMark.clear();
+        $scope.checkForBasicErrors = function() {
```

Comment:
Please could you add a comment explaining what a 'basic error' is (as opposed to a non-basic error)?